Controller/Adminhtml/Menu/MassDelete.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Snowdog\Menu\Controller\Adminhtml\Menu;
+
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Ui\Component\MassAction\Filter;
+use Snowdog\Menu\Model\ResourceModel\Menu\CollectionFactory;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Backend\Model\View\Result\Redirect;
+use Magento\Framework\Exception\LocalizedException;
+use Exception;
+
+/**
+ * Class MassDelete
+ *
+ * This action deletes multiple menus
+ */
+class MassDelete extends Action implements HttpPostActionInterface
+{
+    /**
+     * Authorization level of a basic admin session
+     *
+     * @see _isAllowed()
+     */
+    const ADMIN_RESOURCE = 'Snowdog_Menu::menus';
+
+    /**
+     * @var Filter
+     */
+    protected $filter;
+
+    /**
+     * @var CollectionFactory
+     */
+    protected $collectionFactory;
+
+    /**
+     * @param Context $context
+     * @param Filter $filter
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        Context $context,
+        Filter $filter,
+        CollectionFactory $collectionFactory
+    ) {
+        $this->filter = $filter;
+        $this->collectionFactory = $collectionFactory;
+        parent::__construct($context);
+    }
+
+    /**
+     * Execute action
+     *
+     * @return Redirect
+     * @throws LocalizedException|Exception
+     */
+    public function execute()
+    {
+        $collection = $this->filter->getCollection($this->collectionFactory->create());
+        $collectionSize = $collection->getSize();
+
+        foreach ($collection as $menu) {
+            $menu->delete();
```

Comment:
Please use `\Snowdog\Menu\Api\MenuRepositoryInterface`'s `delete` as `delete` from `AbstractModel` is deprecated.

Reply:
Done